core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -548,7 +548,17 @@ public void removeInstances(Long stackId, Long workspaceId, Set<String> instance
             InstanceMetaData metaData = validateInstanceForDownscale(instanceId, stack, workspaceId, user);
             instanceIdsByHostgroupMap.computeIfAbsent(metaData.getInstanceGroupName(), s -> new LinkedHashSet<>()).add(metaData.getPrivateId());
         }
-        flowManager.triggerStackRemoveInstances(stackId, instanceIdsByHostgroupMap);
+        flowManager.triggerStackRemoveInstances(stackId, instanceIdsByHostgroupMap, false);
+    }
+
+    public void removeInstances(Stack stack, Long workspaceId, List<String> instanceIds, boolean forced, User user) {
```

Comment:
most part of this method is duplicating `public void removeInstances(Long stackId, Long workspaceId, Set<String> instanceIds, User user)`. the common part should be moved out to a separate method

Reply:
ok